cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -53,8 +53,19 @@ class BaseTestStringKernels : public ::testing::Test {
   std::shared_ptr<DataType> offset_type() {
     return TypeTraits<OffsetType>::type_singleton();
   }
+  std::shared_ptr<DataType> list_type();
 };
 
+template <>
+std::shared_ptr<DataType> BaseTestStringKernels<LargeStringType>::list_type() {
+  return large_list(this->type());
```

Comment:
I don't think that makes sense. Why would you return a large list when the embedded type is a large string?

Reply:
because it may have a lot of elements, leading to a long list. But you are right, all cases are possible, e.g. a large string with just a few really large elements, or a normal string that will split into more elements.
Any idea what to do? this made the most sense to me.

Reply:
I think this is overkill for now. Let's just always return a regular list. Large lists will have a sizeable overhead, especially with many small splits.
If there's a user request for it, we may later add an option to return a large list instead.